src/sinks/amqp/tests.rs

```diff
@@ -0,0 +1,228 @@
+use super::*;
+
+#[test]
+pub fn generate_config() {
+    crate::test_util::test_generate_config::<AmqpSinkConfig>();
+}
+
+#[cfg(feature = "amqp-integration-tests")]
+mod integration_tests {
```

Comment:
Kind of a nit but this just occurred to me now but it seems a bit odd or against the common pattern I've seen to have the file tests.rs contain unit tests and a module integration_tests ? Maybe there are examples I haven't seen but, I'm used to seeing the unit tests either in the same file as what is being tested (ie `generate_config()` in the config.rs file) , or have separate files tests.ts and integration_tests.rs. As it stands the namespace for the integration tests to run will look a little redundant : `<...>amqp::tests::integration_tests::<...>`

Reply:
It's definitely been common to split `tests` and `integration_tests` into different modules, it does feel like unnecessary boilerplate if the `tests` file only has the generate configuration fn.

Reply:
Yeah, I'd kind of vote to put the `generate_config()` unit test in the config.rs file, and rename this file to "integration_tests.rs"